examples.js

```diff
@@ -0,0 +1,107 @@
+function examples() {
+
+  setTimeout(blastOff, 86400000);
+
+
+  const locations = ["Austin", "New York", "San Francisco"];
+  locations.forEach(l => {
+    doStuff();
+    doSomeOtherStuff();
+    dispatch(l);
+  });
+
+
+  function createMenu(title, body, buttonText, cancellable) {
+    // ...
+  }
+  
+  createMenu("Foo", "Bar", "Baz", true);
+
+
+  function emailClients(clients) {
+    clients.forEach(client => {
+      const clientRecord = database.lookup(client);
+      if (clientRecord.isActive()) {
+        email(client);
+      }
+    });
+  }
+
+
+  const date = new Date();
+  addToDate(date, 1);
+
+}
+
+
+function showDeveloperList(developers) {
+  developers.forEach(developer => {
+    const expectedSalary = developer.calculateExpectedSalary();
+    const experience = developer.getExperience();
+    const githubLink = developer.getGithubLink();
+    const data = {
+      expectedSalary,
+      experience,
+      githubLink
+    };
+
+    render(data);
+  });
+}
+
+function showManagerList(managers) {
+  managers.forEach(manager => {
+    const expectedSalary = manager.calculateExpectedSalary();
+    const experience = manager.getExperience();
+    const portfolio = manager.getMBAProjects();
+    const data = {
+      expectedSalary,
+      experience,
+      portfolio
+    };
+
+    render(data);
+  });
+}
+
+
+function createFile(name, temp) {
+  if (temp) {
+    fs.create(`./temp/${name}`);
+  } else {
+    fs.create(name);
+  }
+}
+
+
+if (fsm.state === "fetching" && isEmpty(listNode)) {
```

Comment:
Sí la condición se pone muy compleja podrías encapsularla en una función que explique lo que hace
```suggestion
function shouldShowSpinner(fsm, listNode) {
  return fsm.state === "fetching" && isEmpty(listNode);
}

if (shouldShowSpinner(fsmInstance, listNodeInstance)) {...}
```